Trim Appended Filename #17

neilholcomb opened the pull request:
Add new option to use file names relative to the project root.

New option `useFileNameRelativeToProjectRoot`  <-- naming things is hard open to a better name here.

Option is `false` by default to not alter existing behavior.  Again open to changing this so that it is true is the default behavior.  Or not even have an option here???

I have tested both by using babel-cli, as well as inside an ejected create react app...However more testing would be a good idea.  I am also not familiar with testing babel plugins and could use some help/advice for adding test cases to this code.

Commit:
Trim Appended Filename #17

Add new option to use file names relative to the project root.

Commit:
Merge branch 'master' into issues/#17-Trim_Appended_Filename

Commit:
added trailing comma

Commit:
Merge branch 'issues/#17-Trim_Appended_Filename' of https://github.com/mntbkr/babel-plugin-captains-log into issues/#17-Trim_Appended_Filename

Commit:
updating snapshots to reflect new default option

kwelch commented on src/index.js:
```diff
@@ -70,6 +69,33 @@ export default function({ types: t }) {
       },
     },
   };
+  function getFileName( file, options ) {
```
I would write a specific test on how this function works. Should be good since it is a pure function, it will need to be exported. 

I would like to see end to end tests also, but I am not sure how to get path data passed in to our tests. I will research that.

Commit:
Updated path code

using paths relative to project dir is expected behavior now
updated snapshots to reflect option being removed
added test cases for getFileName.js

Commit:
added missing commas

mntbkr commented:
I have moved getFileName into its own file under utils.  And added some basic tests for this method.

The relative to project root path is now the expected default behavior when using injectFileName

I have tested this against a couple of my own projects by using `yarn link` I would still recommend you test against your projects before merging.

kwelch commented on src/utils/getFileName.spec.js:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    expect( getFileName(file) ).toEqual('./path/to/file.js');
+  })
```
Looks good can we add one where the path is really long and it is trimmed based on a custom PWD?

kwelch commented:
I will take a look at this with a project of mine at some point this weekend. Thanks

kwelch commented:
Could you all add your self to the `all-contributors`?

Commit:
added name to all-contributorsrc

kwelch commented:
I haven't forgotten about this. It is chill month for me so I am a little slower than normal.

neilholcomb commented:
No problem...when the time comes let me know if there is anything else I can do to help land this PR

kwelch commented:
Is this something you need? If it works for you and you are hoping to use it. I am good with merging based on what I have seen.

neilholcomb commented:
No, sadly I don't have access to babel in my day job :-( ....this was just an excercise for me to learn more about babel plugins.  Merge at your leisure.

kwelch commented:
I appreciate the change and I hope it helped with your learning exercise I appreciate it.

kwelch commented:
I am trying this out but it is not working as I expected. I am playing with your branch. I will push some changes as I look at it.

peteringram0 commented:
Is this likely to ever make it into the project at all? Thanks.

kwelch commented:
I am still open to this feature. If you have a recommendation on the solution I would love some help solving it. Last time I worked on it, I ran into some issues.
